compiler/deSugar/Match.hs

```diff
@@ -228,70 +230,73 @@ match vars@(v:_) ty eqns    -- Eqns *can* be empty
           maybeWarn $ (map (\g -> text "Putting these view expressions into the same case:" <+> (ppr g))
                        (filter (not . null) gs))
 
-matchEmpty :: HasCallStack => MatchId -> Type -> DsM [MatchResult]
+matchEmpty :: HasCallStack => MatchId -> Type -> Rig -> DsM [MatchResult]
 -- See Note [Empty case expressions]
-matchEmpty var res_ty
+matchEmpty var res_ty weight
   = return [MatchResult CanFail mk_seq]
   where
     mk_seq fail = return $ mkWildCase (Var var) (Weighted (idWeight var) (idType var)) res_ty
                                       [(DEFAULT, [], fail)]
 
-matchVariables :: [MatchId] -> Type -> [EquationInfo] -> DsM MatchResult
+matchVariables :: [MatchId] -> Type -> Rig -> [EquationInfo] -> DsM MatchResult
 -- Real true variables, just like in matchVar, SLPJ p 94
 -- No binding to do: they'll all be wildcards by now (done in tidy)
-matchVariables (_:vars) ty eqns = match vars ty (shiftEqns eqns)
-matchVariables [] _ _ = panic "matchVariables"
+matchVariables (_:vars) ty weight eqns = match vars ty weight(shiftEqns eqns)
+matchVariables [] _ _ _ = panic "matchVariables"
 
-matchBangs :: [MatchId] -> Type -> [EquationInfo] -> DsM MatchResult
-matchBangs (var:vars) ty eqns
-  = do  { match_result <- match (var:vars) ty $
+matchBangs :: [MatchId] -> Type -> Rig -> [EquationInfo] -> DsM MatchResult
+matchBangs (var:vars) ty weight eqns
+  = do  { match_result <- match (var:vars) ty weight $
                           map (decomposeFirstPat getBangPat) eqns
         ; return (mkEvalMatchResult var ty match_result) }
-matchBangs [] _ _ = panic "matchBangs"
+matchBangs [] _ _ _ = panic "matchBangs"
 
-matchCoercion :: HasCallStack => [MatchId] -> Type -> [EquationInfo] -> DsM MatchResult
+matchCoercion :: HasCallStack => [MatchId] -> Type -> Rig -> [EquationInfo] -> DsM MatchResult
 -- Apply the coercion to the match variable and then match that
-matchCoercion (var:vars) ty (eqns@(eqn1:_))
+matchCoercion (var:vars) ty weight (eqns@(eqn1:_))
   = do  { let CoPat _ co pat _ = firstPat eqn1
         ; let pat_ty' = hsPatType pat
-        ; var' <- newUniqueId var (idWeight var) pat_ty'
-        ; match_result <- match (var':vars) ty $
+        ; var' <- newUniqueId var Omega pat_ty'
```

Comment:
Is `Omega` correct here? If so, I suggest making a comment explaining why.

Reply:
No! I think that is a mistake.